### PR TITLE
[SYCL-MLIR]: Runtime workaround for paralllel_for

### PR DESCRIPTION
**Pull request by etiotto**
The following test case (using a SYCL parallel_for) fails to run correctly due to a SYCL RT function (dim_loop_impl) which uses a complex templated implementation which "confuses" cgeist. As a workaround we can use the older SYCL RT implementation which used a regular for loop.  

Test case: 
```
void host_parallel_for(std::array<int, N> &A) {
  auto q = queue{};
  device d = q.get_device();
  std::cout << "Using " << d.get_info<info::device::name>() << "\n";
  auto range = sycl::range<1>{N};

  {
    auto buf = buffer<int, 1>{A.data(), range};
    q.submit([&](handler &cgh) {
      auto A = buf.get_access<access::mode::write>(cgh);
      cgh.parallel_for<class kernel_parallel_for>(
          range, [=](sycl::id<1> id) { A[3] = 33; });
    });
  }
}

### CODE REVIEW

**Comment by etiotto** [comment]
Note: The original clang-format for the RT file I modifies was incorrect. I did not want to push changes related to code formatting so that is the reason the linting job fails.

File: sycl/include/sycl/accessor.hpp [hunk 1]
```diff
@@ -2458,7 +2466,8 @@ class __SYCL_SPECIAL_CLASS local_accessor_base :
 
 #endif // __SYCL_DEVICE_ONLY__
 
-  // Method which calculates linear offset for the ID using Range and Offset.
+  // Method which calculates linear offset for the ID using Range and
```
Comment (whitneywhtsang) [inline sycl/include/sycl/accessor.hpp:2469]: Please revert all clang-format changes.

Comment (etiotto) [reply sycl/include/sycl/accessor.hpp]: oops I still have some leftover clang-format changes

**Review by victor-eds — APPROVED** [review]
LGTM

**Review by whitneywhtsang — APPROVED** [review]

**Comment by victor-eds** [comment]
@etiotto Pipeline is failing here

**Comment by whitneywhtsang** [comment]
> @etiotto Pipeline is failing here

Fixed by https://github.com/intel/llvm/pull/7306.

**Pull request merged**